common/web/lm-worker/src/correction/distance-modeler.ts

```diff
@@ -721,6 +730,14 @@ namespace correction {
         timer.startLoop();
         while(preprocessedQueue.count > 0) {
           let entry = preprocessedQueue.dequeue();
+
+          // Is the entry a reasonable result?
+          if(entry.isFullReplacement) {
```

Comment:
If there are a great many high-frequency words on a nearby key, will we run out of search results before we get to low-frequency words on the preferred key?

Reply:
With the way the correction-search algorithm works, we won't have to worry about this for entries of a single keystroke.

This class (and its algorithm) are all about finding good corrections without the need to "look ahead" or predict.  Our Trie setup allows us to know the "weight" of each of its branches without having to traverse down them.

We only enumerate over the "great many... words on a nearby key" on a later layer _after_ we have identified good corrections.  We only enumerate 12 or so for each correction / "prediction root", as we only return 12 total suggestions - any extras would automatically lose, after all.  Then, we combine all sets and rank 'em to determine what wins.

So, the most likely key will _always_ be the top-ranked "correction" / "prediction root", and thus it will _always_ be selected within its layer.  After all, the most likely thing is to keep what we already have.  It's just that other "corrections" may have higher-frequency words.